docs/admin/security_event_logs.mdx

```diff
@@ -0,0 +1,130 @@
+- This function takes following information to create a log event
+  - Context contains information on the acting user
+  - SecurityEventName which is predefined [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/security_event_logs.go?L22-101)
+  - URL if available
+  - userID of the user that the action is applied towards
+  - anonymousUserID for unauthenticated users
+  - source of the log
+  - arguments relevant to the action being logged
+- Example of using the function to logan event
+  ```go
+  db.SecurityEventLogs().LogSecurityEvent(ctx, database.SecurityEventNameEmailAdded, r.URL.Path, uint32(actr.UID), "", "BACKEND", email)
+
+- The function sends the log event it creates to be pushed to the right location based on the site-config settings
+- The function also checks to make sure that marshaling the arguments does not cause as error
+
+## How to find security events in logs
+- Security events are logged with all the relevant information associated with the actions
+- Depending on the location of the log destination set in the site config, the event log can be either found in the application log output or in the database or both.
+- A sample output of a logged event from application logs would look similar to this:
+  ```JSON
+  {
+  ...
+  {
+    "message": "EmailAdded (sampling immunity token: 12345-222-3333-5454-9w08fs0s9d8f)",
+    "Caller": "audit/audit.go:57",
+    "Attributes": {
+      "audit": {
+        "actor": {
+          "X-Forwarded-For": "127.0.0.1",
+          "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) ",
+          "ip": "100.211.3.155",
+          "actorUID": "123245"
+        },
+        "entity": "security events",
+        "auditId": "12345-222-3333-5454-9w08fs0s9d8f",
+        "action": "EmailAdded"
+      },
+      "event": {
+        "URL": "",
+        "argument": "\"new@sourcegraph.com\"",
+        "AnonymousUserID": "",
+        "UserID": 223955,
+        "source": "12345",
+        "timestamp": "2023-12-21 02:41:08.649603776 +0000 UTC",
+        "version": "255367_2023-12-20_5.2-a3143120c41e"
+      }
+    },
+    "Function": "github.com/sourcegraph/sourcegraph/internal/audit.Log",
+    "InstrumentationScope": "frontend.SecurityEvents",
+    "timestampNanos": 1703126468649641000,
+    "Resource": {
+      "service.instance.id": "sourcegraph-frontend-769bdbdd77-p2f8j",
+      "service.name": "frontend",
+      "service.version": "255367_2023-12-20_5.2-a3143120c41e"
+    }
+  }
+  ...
+  }
+
+- Entity field can be used to filter on all security events.
+- Action field will provide information on the event and can be correlated with the action taken, in this case EmailAdded
+- The actorUID can be used to filter out events from a particular user
+- UserID can be used to filter out actions taken on a particular user's information
+  
+## FAQ
+What events are currently being logged as security events?
```

Comment:
@andreeleuterio is this what you are looking for ? or did you want me to add a link to elastic search query?